content/v2.0/introduction/getting-started.md

```diff
@@ -138,29 +162,1261 @@ Visit your application in the browser at [http://localhost:2300](http://localhos
+  let(:books) { app["persistence.rom"].relations[:books] }
+
+  before do
+    books.insert(title: "Refactoring", author: "Martin Fowler")
+    books.insert(title: "Test Driven Development", author: "Kent Beck")
+  end
+
+  it "returns a list of books" do
+    get "/books"
+
+    expect(last_response).to be_successful
+    expect(last_response.content_type).to eq("application/json; charset=utf-8")
+
+    response_body = JSON.parse(last_response.body)
+
+    expect(response_body).to eq([
+      { "title" => "Refactoring", "author" => "Martin Fowler" },
+      { "title" => "Test Driven Development", "author" => "Kent Beck" }
+    ])
+  end
+end
+```
+
+To get this spec to pass, we'll need to update our books index action to return books from the books relation.
+
+To access the books relation within the action, we can use Hanami's "Deps mixin". Covered in detail in the [containers and components](/v2.0/architecture/containers/) section of the [Architecture guide](/v2.0/architecture/containers/), the Deps mixin gives each of your application's components easy access to the other components it depends on to achieve its work. We'll see this in more detail in a moment.
+
+For now however, it's enough to know that we can use `include Deps["persistence.rom"]` to make rom-rb available via a `rom` method within our action. The books relation is then available via `rom.relations[:books]`.
+
+To satisy our spec, we need to meet a few requirements. Firstly, we want to render each book's _title_ and _author_, but not its _id_. Secondly we want to return books alphabetically by title. We can achieve these requirements using the `select` and `order` methods offered by the books relation:
+
+```ruby
+# frozen_string_literal: true
+
+module Bookshelf
+  module Actions
+    module Books
+      class Index < Bookshelf::Action
+        include Deps["persistence.rom"]
+
+        def handle(*, response)
+          books = rom.relations[:books]
+            .select(:title, :author)
+            .order(:title)
+            .to_a
+
+          response.format = :json
+          response.body = books.to_json
+        end
+      end
+    end
+  end
+end
+```
+
+<p class="convention">
+  Accessing relations directly from actions is not a commonly recommended pattern. Instead, a <a href="https://rom-rb.org/5.0/learn/repositories/quick-start/">rom repository</a> should be used. Here, however, the repository is ommitted for brevity.
+</p>
+
+With this action in place, the spec passes once more:
+
+```shell
+bundle exec rspec spec/requests/books/index_spec.rb
+
+GET /books
+  returns a list of books
+
+Finished in 0.05765 seconds (files took 1.36 seconds to load)
+1 example, 0 failures
+```
+
+## Parameter validation
+
+Of course, returning _every_ book in the database when a visitor makes a request to `/books` is not going to be a good strategy for very long. Luckily rom-rb relations offer pagination support. Let's add pagination with a default page length of 5:
```

Comment:
Is it "page length"? I think I usually see "page size".

Reply:
Yeah, that makes more sense - adjusted to page size.